js/core/template_manager.js

```diff
@@ -0,0 +1,221 @@
+        this._defaultTemplates = {};
+        this._anonymousTemplateName = anonymousTemplateName || ANONYMOUS_TEMPLATE_NAME;
+
+        this._createElement = createElement || defaultCreateElement;
+        this._createTemplateIfNeeded = this._createTemplateIfNeeded.bind(this);
+    }
+
+    static createDefaultOptions() {
+        return {
+            integrationOptions: {
+                watchMethod: (fn, callback, options = {}) => {
+                    if(!options.skipImmediate) {
+                        callback(fn());
+                    }
+                    return noop;
+                },
+                templates: { 'dx-polymorph-widget': DX_POLYMORPH_WIDGET_TEMPLATE },
+            }
+        };
+    }
+
+    get anonymousTemplateName() {
+        return this._anonymousTemplateName;
+    }
+
+    addDefaultTemplates(templates) {
+        this._defaultTemplates = extend({}, this._defaultTemplates, templates);
+    }
+
+    dispose() {
+        this._tempTemplates.forEach(tempTemplate => {
+            tempTemplate.template.dispose && tempTemplate.template.dispose();
+        });
+        this._tempTemplates = [];
+    }
+
+    extractTemplates($el) {
+        const templates = this._extractTemplates($el);
+        const anonymousTemplateMeta = this._extractAnonymousTemplate($el);
+        return { templates, anonymousTemplateMeta };
+    }
+
+    _extractTemplates($el) {
+        const templateElements = $el.contents().filter(TEMPLATE_SELECTOR);
+        const templatesMap = {};
+
+        templateElements.each((_, template) => {
+            const templateOptions = getElementOptions(template).dxTemplate;
+
+            if(!templateOptions) {
+                return;
+            }
+
+            if(!templateOptions.name) {
+                throw Error('E0023');
+            }
+
+            $(template).addClass(TEMPLATE_WRAPPER_CLASS).detach();
+            templatesMap[templateOptions.name] = templatesMap[templateOptions.name] || [];
+            templatesMap[templateOptions.name].push(template);
+        });
+
+        const templates = [];
+        for(let templateName in templatesMap) {
+            const deviceTemplate = findTemplateByDevice(templatesMap[templateName]);
+            if(deviceTemplate) {
+                templates.push({
+                    name: templateName,
+                    template: this._createTemplate(deviceTemplate),
+                });
+            }
+        }
+        return templates;
+    }
+
+    _extractAnonymousTemplate($el) {
+        const $anonymousTemplate = $el.contents().detach();
+
+        const $notJunkTemplateContent = $anonymousTemplate.filter((_, element) => {
+            const isTextNode = element.nodeType === TEXT_NODE;
+            const isEmptyText = $(element).text().trim().length < 1;
+
+            return !(isTextNode && isEmptyText);
+        });
+        const onlyJunkTemplateContent = $notJunkTemplateContent.length < 1;
+
+        return !onlyJunkTemplateContent
```

Comment:
```suggestion
        return $notJunkTemplateContent.length > 0
```
Seems to be clear enough